rpc.go

```diff
@@ -247,15 +202,40 @@ func (r *rpc) Clear(in *kvv1.Request, _ *kvv1.Response) error {
 	return nil
 }
 
-func from(tr []*kvv1.Item) []kv.Item {
-	items := make([]kv.Item, 0, len(tr))
+func (r *rpc) getStorage(in *kvv1.Request) (kv.Storage, error) {
+	if in.GetStorage() == "" {
+		return nil, errors.Str("no storage provided")
+	}
+
+	storage, ok := r.storages[in.GetStorage()]
+
+	if !ok {
+		return nil, errors.Errorf("no such storage: %s", in.GetStorage())
+	}
+
+	return storage, nil
+}
+
+func composeKeys(in *kvv1.Request) *[]string {
```

Comment:
No, this is the same as the previous approach. You may check this tutorial: https://go101.org/article/container.html
If you know len, you may pass an already allocated slice with the len like make([]string, len). Then add elements inside the func, and since slices are basically fat pointers, outside slice would be filled with elements. Instead of allocating slice in the function. This is a small optimization, but since KV methods are supposed to be hot, I'd prefer to slightly optimize allocations here.